addons/utm/models/utm_source.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import _, api, fields, models, tools
+
+
+class UtmSource(models.Model):
+    _name = 'utm.source'
+    _description = 'UTM Source'
+
+    name = fields.Char(string='Source Name', required=True)
+
+    _sql_constraints = [
+        ('unique_name', 'UNIQUE(name)', 'The name must be unique'),
+    ]
+
+    @api.model
+    def create(self, vals):
+        vals['name'] = self.env['utm.mixin']._generate_unique_name(self, vals.get('name'))
+        return super(UtmSource, self).create(vals)
+
+    def _generate_name(self, record, content):
```

Comment:
Why don't we have the same for medium ? Because no use currently ? It could be available on the mixin, along with _generate_unique_name . That way with a slight renaming we could have
```
_generate_unique_name_based_on_content
_set_name_unique
```
which would be clearer to people I think.